server/util/async.ts

```diff
@@ -0,0 +1,8 @@
+export async function asyncFilter<T>(
+  arr: Array<T>,
+  predicate: (item: T, index: number) => Promise<boolean>,
+): Promise<Array<T>> {
+  const results = await Promise.all(arr.map(predicate));
```

Comment:
should use for loop and await to avoid too many promises maybe?

Reply:
Yeah. That's a good idea.

Reply:
then it's already done...